run_tests.py

```diff
@@ -114,34 +117,74 @@ def test_from_failstr(failstr):
-disfile = os.path.dirname(__file__) + DISABLED_TESTS_FILE_NAME
-
-# this file is needed for tests with TCP segmentation
-DISABLED_TESTS_FILE_NAME_TCP_SEG = "/tests_disabled_tcpseg.json"
-disfile_tcp_seg = os.path.dirname(__file__) + DISABLED_TESTS_FILE_NAME_TCP_SEG
+def __check_memory_consumption(python_memory_before_: int, used_memory_before_: int) -> None:
+    """Check a memory consumption after all tests."""
+    if run_config.CHECK_MEMORY_LEAKS:
+        gc.collect()
+        time.sleep(1)
+        python_memory_after = psutil.Process().memory_info().rss // 1024
+        delta_python = python_memory_after - python_memory_before_
+
+        used_memory_after = util.get_used_memory()
+        memleak_msg = (
+            "----------------------------------------------------------------------------\n"
+            "Check memory leaks for test suite:\n"
+            f"Before: used memory: {used_memory_before_};\n"
+            f"Before: python memory: {python_memory_before_};\n"
+            f"After: used memory: {used_memory_after};\n"
+            f"After: python memory: {python_memory_after};\n"
+            "----------------------------------------------------------------------------\n"
+        )
+        tf_cfg.dbg(1, memleak_msg)
+        assert (
+            used_memory_after - delta_python - used_memory_before_
+            <= run_config.MEMORY_LEAK_THRESHOLD
+        ), memleak_msg
+
+
+def __check_kmemleak() -> None:
+    """Check kmemleak result if `--kernel-dbg` option is present."""
+    if run_config.KERNEL_DBG_TESTS and KMEMLEAK_IS_PRESENT:
+        # we should run TempestaFW again to display the output correctly
+        tempesta = control.Tempesta(vhost_auto=False)
+        tempesta.config.set_defconfig("")
+        tempesta.disable_check_config()
+        tempesta.start()
+
+        try:
+            remote.tempesta.run_cmd("echo scan > /sys/kernel/debug/kmemleak", timeout=60)
+            stdout, stderr = remote.tempesta.run_cmd("cat /sys/kernel/debug/kmemleak", timeout=60)
+        finally:
+            tempesta.stop()
+        kmemleak_msg = (
+            "kmemleak result.\n"
+            "----------------------------------------------------------------------------\n"
+            f"{stdout.decode()}\n"
+            "----------------------------------------------------------------------------\n"
+        )
+        tf_cfg.dbg(1, kmemleak_msg if stdout else "/sys/kernel/debug/kmemleak is empty")
+        assert b"tfw_" not in stdout, kmemleak_msg
 
-# this file is needed for tests with remote config
-DISABLED_TESTS_FILE_NAME_REMOTE = "/tests_disabled_remote.json"
-disfile_remote = os.path.dirname(__file__) + DISABLED_TESTS_FILE_NAME_REMOTE
 
-TESTS_PRIORITY_FILE_NAME = "/tests_priority"
-priority_file = os.path.dirname(__file__) + TESTS_PRIORITY_FILE_NAME
-t_priority_out = open(priority_file).readlines()
+t_priority_out = open(os.path.dirname(__file__) + "/tests_priority").readlines()
```

Comment:
We use different (all possible) string formatting/concatenations technique. I would choose one and use it for the code base - it is best practice.

For this particular case, the best practice to concatenate path components is using `os.path.join` for reasons. It related for all same cases. For the future.

Reply:
Done (with `os.path.join`). Currently we use `f` for string concatenations